connectors/filtering/validation.py

```diff
@@ -28,15 +29,21 @@ class SyncRuleValidationResult:
 
     ADVANCED_RULES = "advanced_snippet"
 
-    def __init__(self, rule_id, is_valid, validation_message):
-        self.rule_id = rule_id
+    def __init__(self, rule_ids, is_valid, validation_message):
+        # `str` and `bytes` are also iterable
+        if not isinstance(rule_ids, Iterable) or isinstance(rule_ids, (str, bytes)):
+            rule_ids = [rule_ids]
+        else:
+            rule_ids = list(rule_ids)
```

Comment:
I think this is okish for now, as we sometimes use the `rule_ids` keyword explicitly and sometimes we omit it. So this makes sure that everything is working without scanning through every usage of `SyncRuleValidationResult`. Maybe we should revisit using type hints again.